### PR TITLE
LPS-145001 Date validation is crashing when changing form language

### PR DESCRIPTION
**Pull request by mateusgomes01**
Related Issue: https://issues.liferay.com/browse/LPS-145001

### CODE REVIEW

**Comment by liferay-forms** [comment]
Hi, you don’t need to run the CI commands anymore, I’ll do it for you and let you know if any issue happens.

**Comment by liferay-forms** [comment]
ci:test:sf

**Comment by liferay-forms** [comment]
ci:test:relevant

**Comment by liferay-forms** [comment]
Hey @mateusgomes01, can you please check if these failures are related to your changes? Thanks

**Comment by mateusgomes01** [comment]
ci:test:relevant

**Comment by liferay-forms** [comment]
Hey @mateusgomes01, can you please check if these failures are related to your changes? Thanks

**Comment by mateusgomes01** [comment]
ci:test:sf

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/Validation/Validation.js [hunk 1]
```diff
@@ -120,7 +120,9 @@ const Validation = ({
 					name={name}
 					onBlur={onBlur}
 					parameter={parameter}
-					parentFieldName={validation.fieldName}
+					parentFieldName={
+						validation?.fieldName ?? focusedField?.fieldName
+					}
```
Comment (mateusgomes01) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/Validation/Validation.js:125]: @miltonmc Not the solution we desired, but we're blocked by the inconsistent behavior of the focusedField state

**Comment by liferay-forms** [comment]
ci:test:relevant

**Comment by liferay-forms** [comment]
Hey @mateusgomes01, can you please check if these failures are related to your changes? Thanks

**Comment by mateusgomes01** [comment]
ci:reevaluate:1277401_1384

**Comment by liferay-forms** [comment]
Hey @mateusgomes01, can you please check if these failures are related to your changes? Thanks

**Comment by mateusgomes01** [comment]
ci:reevaluate:1277401_1384

**Comment by liferay-forms** [comment]
Hey @mateusgomes01, can you please check if these failures are related to your changes? Thanks

**Comment by mateusgomes01** [comment]
ci:reevaluate:1277401_1384

**Comment by liferay-forms** [comment]
Hey @mateusgomes01, can you please check if these failures are related to your changes? Thanks

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/Validation/Validation.js (hunk 1 above)
Comment (miltonmc) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/Validation/Validation.js:125]: It's ok as a quick fix, but we should sign that this solution should not be replicated.

Please extract this code  and the similar one in the reducer to a variable and add a `TODO` explaining why we did that and what we should expect instead:
```javascript
/* TODO: Remove this variable when the fieldName is properly delivered by the validator prop */
const parentFieldName = validation?.fieldName ?? focusedField.fieldName;
```

**Review by miltonmc — CHANGES_REQUESTED** [review]

**Comment by miltonmc** [comment]
Please always clean up your PRs in order to contains only the commits you implemented.
It helps me to do a quicker review. Thanks! :smile:

**Comment by mateusgomes01** [comment]
ci:test:sf

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/Validation/Validation.js [hunk 2]
```diff
@@ -120,7 +122,11 @@ const Validation = ({
 					name={name}
 					onBlur={onBlur}
 					parameter={parameter}
-					parentFieldName={validation.fieldName}
+
+					/* TODO: Remove the focusedField variable when the fieldName is properly delivered by the validator prop */
+					parentFieldName={
+						validation?.fieldName ?? focusedField?.fieldName
+					}
```
Comment (miltonmc) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/Validation/Validation.js:129]:
```suggestion
					parentFieldName={parentFieldName}
```

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/Validation/Validation.js [hunk 3]
```diff
@@ -60,6 +60,8 @@ const Validation = ({
 	] = useReducer(
 		reducer({
 			editingLanguageId,
+
+			/* TODO: Remove the focusedField variable when the fieldName is properly delivered by the validator prop */
 			fieldName: validation?.fieldName ?? focusedField?.fieldName,
```
Comment (miltonmc) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/Validation/Validation.js:65]:
```javascript
/* TODO: Remove this variable when the fieldName is properly delivered by the validator prop */
const parentFieldName = validation?.fieldName ?? focusedField.fieldName;
```

```suggestion
			fieldName: parentFieldName,
```

**Review by miltonmc — CHANGES_REQUESTED** [review]

**Comment by mateusgomes01** [comment]
ci:test:sf

**Comment by liferay-forms** [comment]
Hey @mateusgomes01, can you please check if these failures are related to your changes? Thanks

**Comment by mateusgomes01** [comment]
ci:test:relevant

**Comment by miltonmc** [comment]
ci:forward

**Pull request closed without merging**